### DISCUSSION TITLE
KPN iptv stopped working (UDM-pro-SE)

### DISCUSSION THREAD

**Discussion by Pamaro1976** [fabianishere/udm-iptv#164] [Q&A]
=======
update 2-9-2022: 
The issue grew bigger getting network errors and losing internet connectivity. Did additional checks using the original Experia box. Problem turned out to be a KPN problem. The FTU failed (gen 1 models have dodgy ethernet ports). FTU got replaced with a gen 2, everything came up as it should ;-)

Question can be closed.
=======

Since this afternoon my IPTV stopped working (directly connected to the FTU)

Checked the config and the debug. The Arris 5202 keeps displaying error 563. Did a reset on the receiver

**Steps taken:**
1: reboot iptv receiver
2: reinstalled iptv receiver (factory default)
3: removed igmp packages
4: installed latest igmp packages
5: reboot udm-pro-se (UniFi OS UDM SE 2.5.11)
6: reboot iptv receiver
7: reinstalled iptv receiver (factory default)
8: checked with laptop in iptv VLAN (DHCP address issued)
9: ARRIS selftest shows "platform unavailable"

**output of "route list" and "VLAN interface"**

_ip route list_
"removed" dev br0 proto kernel scope link src "removed"
192.168.3.0/24 dev br4 proto kernel scope link src 192.168.3.1 
"removed" dev ppp0 proto kernel scope link src "removed"

_ip -4 addr show dev br4_
33: br4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 192.168.3.1/24 scope global br4
       valid_lft forever preferred_lft forever
       
**output of "journalctl"**

journalctl -u udm-iptv | grep 20:19:

Aug 29 20:19:02 Pamaro-FW udm-iptv[5409]: Sending discover...
Aug 29 20:19:05 Pamaro-FW udm-iptv[5409]: Sending discover...
Aug 29 20:19:08 Pamaro-FW udm-iptv[5409]: udhcpc failed to get a DHCP lease
Aug 29 20:19:08 Pamaro-FW udm-iptv[5409]: No lease, forking to background
Aug 29 20:19:08 Pamaro-FW udm-iptv[5409]: NATing IPTV network ranges (if necessary)
Aug 29 20:19:08 Pamaro-FW udm-iptv[5409]: Setting up igmpproxy
Aug 29 20:19:08 Pamaro-FW udm-iptv[5409]: Starting igmpproxy
Aug 29 20:19:08 Pamaro-FW igmpproxy[5409]: There must be at least 1 Vif as upstream.
Aug 29 20:19:08 Pamaro-FW systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Aug 29 20:19:08 Pamaro-FW systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Aug 29 20:19:08 Pamaro-FW systemd[1]: udm-iptv.service: Consumed 144ms CPU time.
Aug 29 20:19:13 Pamaro-FW systemd[1]: udm-iptv.service: Service RestartSec=5s expired, scheduling restart.
Aug 29 20:19:13 Pamaro-FW systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 118.
Aug 29 20:19:13 Pamaro-FW systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Aug 29 20:19:13 Pamaro-FW systemd[1]: udm-iptv.service: Consumed 144ms CPU time.
Aug 29 20:19:13 Pamaro-FW systemd[1]: Started IPTV support for the UniFi Dream Machine.
Aug 29 20:19:13 Pamaro-FW udm-iptv[6142]: Device iptv already exists.. Deleting device
Aug 29 20:19:14 Pamaro-FW udm-iptv[6142]: Obtaining IP address for VLAN interface
Aug 29 20:19:14 Pamaro-FW udm-iptv[6142]: udhcpc (v1.22.1) started
Aug 29 20:19:14 Pamaro-FW udm-iptv[6142]: Sending discover...
Aug 29 20:19:17 Pamaro-FW udm-iptv[6142]: Sending discover...
Aug 29 20:19:20 Pamaro-FW udm-iptv[6142]: Sending discover...

**Comment by fabianishere** [comment]
It looks like you are not receiving an IP from KPN for the IPTV network. Could you share the output of `udm-iptv-diag`?

**Reply by Pamaro1976** [reply]
**systemctl status udm-iptv**
● udm-iptv.service - IPTV support for the UniFi Dream Machine
   Loaded: loaded (/lib/systemd/system/udm-iptv.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2022-08-29 20:42:04 CEST; 3s ago
 Main PID: 12891 (udm-iptv)
    Tasks: 2 (limit: 4726)
   Memory: 868.0K
      CPU: 26ms
   CGroup: /system.slice/udm-iptv.service
           ├─12891 /bin/sh /usr/lib/udm-iptv/udm-iptv start
           └─12901 udhcpc -b -R -p /var/run/udhcpc.iptv.pid -s /usr/lib/udm-iptv/udhcpc.hook -i iptv -O staticroutes -V IPTV_RG

Aug 29 20:42:04 Pamaro-FW systemd[1]: Started IPTV support for the UniFi Dream Machine.
Aug 29 20:42:04 Pamaro-FW udm-iptv[12891]: Device iptv already exists.. Deleting device
Aug 29 20:42:04 Pamaro-FW udm-iptv[12891]: Obtaining IP address for VLAN interface
Aug 29 20:42:04 Pamaro-FW udm-iptv[12891]: udhcpc (v1.22.1) started
Aug 29 20:42:04 Pamaro-FW udm-iptv[12891]: Sending discover...
Aug 29 20:42:07 Pamaro-FW udm-iptv[12891]: Sending discover...

**udm-iptv-diag**
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: br4
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
=== Service Logs ===
Aug 29 20:44:31 Pamaro-FW systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 221.
Aug 29 20:44:31 Pamaro-FW systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Aug 29 20:44:31 Pamaro-FW systemd[1]: udm-iptv.service: Consumed 111ms CPU time.
Aug 29 20:44:31 Pamaro-FW systemd[1]: Started IPTV support for the UniFi Dream Machine.
Aug 29 20:44:31 Pamaro-FW udm-iptv[20243]: Device iptv already exists.. Deleting device
Aug 29 20:44:31 Pamaro-FW udm-iptv[20243]: Obtaining IP address for VLAN interface
Aug 29 20:44:31 Pamaro-FW udm-iptv[20243]: udhcpc (v1.22.1) started
Aug 29 20:44:31 Pamaro-FW udm-iptv[20243]: Sending discover...
Aug 29 20:44:34 Pamaro-FW udm-iptv[20243]: Sending discover...
Aug 29 20:44:38 Pamaro-FW udm-iptv[20243]: Sending discover...

**Reply by fabianishere** [reply]
The cable from your FTU is connected to Port 9 (the RJ45 port) on your UDM Pro, right?

**Reply by Pamaro1976** [reply]
That is correct. the setup worked fine till today (been using it since March 2022).

**Reply by fabianishere** [reply]
There might be an issue at KPN's side (since it is not responding to the DHCP requests) or there's some firewall rule blocking the outgoing requests.

**Reply by Pamaro1976** [reply]
I will check, but there are no additional Firewall rules added since March. I did put in a portforwarding for an OpenVPN server on Sunday (iptv was working that evening).

**Reply by Pamaro1976** [reply]
Just noticed this message (after issueing the  "systemctl status udm-iptv" several times in a row)

### systemctl status udm-iptv
● udm-iptv.service - IPTV support for the UniFi Dream Machine
   Loaded: loaded (/lib/systemd/system/udm-iptv.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Mon 2022-08-29 21:04:53 CEST; 4s ago
  Process: 17306 ExecStart=/usr/lib/udm-iptv/udm-iptv start (code=exited, status=255/EXCEPTION)
 Main PID: 17306 (code=exited, status=255/EXCEPTION)
      CPU: 138ms

Aug 29 21:04:53 Pamaro-FW systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Aug 29 21:04:53 Pamaro-FW systemd[1]: udm-iptv.service: Consumed 138ms CPU time.

### systemctl status udm-iptv
● udm-iptv.service - IPTV support for the UniFi Dream Machine
   Loaded: loaded (/lib/systemd/system/udm-iptv.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2022-08-29 21:04:58 CEST; 1s ago
 Main PID: 18074 (udm-iptv)
    Tasks: 2 (limit: 4726)
   Memory: 788.0K
      CPU: 26ms
   CGroup: /system.slice/udm-iptv.service
           ├─18074 /bin/sh /usr/lib/udm-iptv/udm-iptv start
           └─18083 udhcpc -b -R -p /var/run/udhcpc.iptv.pid -s /usr/lib/udm-iptv/udhcpc.hook -i iptv -O staticroutes -V IPTV_RG

Aug 29 21:04:58 Pamaro-FW systemd[1]: Started IPTV support for the UniFi Dream Machine.
Aug 29 21:04:58 Pamaro-FW udm-iptv[18074]: Device iptv already exists.. Deleting device
Aug 29 21:04:58 Pamaro-FW udm-iptv[18074]: Obtaining IP address for VLAN interface
Aug 29 21:04:58 Pamaro-FW udm-iptv[18074]: udhcpc (v1.22.1) started
Aug 29 21:04:58 Pamaro-FW udm-iptv[18074]: Sending discover...

**Reply by fabianishere** [reply]
You can use `journalctl -u udm-iptv` to view the complete log.

**Reply by Pamaro1976** [reply]
> journalctl -u udm-iptv

The issue seems to have started around 18:44:23, the last multicast leave was seen and the routes have been removed from the routingtable. It's also the moment the exception message started to appear (18:44:32).

Aug 29 18:44:18 Pamaro-FW udm-iptv[4064]: Updated route entry for 224.0.252.134 on VIF #0
Aug 29 18:44:18 Pamaro-FW udm-iptv[4064]: Adding MFC: 217.166.226.134 -> 224.0.252.134, InpVIf: 1
Aug 29 18:44:21 Pamaro-FW udm-iptv[4064]: RECV V2 member report   from 192.168.3.1     to 224.0.0.251
Aug 29 18:44:21 Pamaro-FW udm-iptv[4064]: The IGMP message was from myself. Ignoring.
Aug 29 18:44:21 Pamaro-FW udm-iptv[4064]: RECV V2 member report   from 192.168.3.1     to 224.0.0.22
Aug 29 18:44:21 Pamaro-FW udm-iptv[4064]: The IGMP message was from myself. Ignoring.
Aug 29 18:44:23 Pamaro-FW systemd[1]: Stopping IPTV support for the UniFi Dream Machine...
Aug 29 18:44:23 Pamaro-FW udm-iptv[4064]: select() failure; Errno(4): Interrupted system call
Aug 29 18:44:23 Pamaro-FW udm-iptv[4064]: Got a interrupt signal. Exiting.
Aug 29 18:44:23 Pamaro-FW udm-iptv[4064]: Leaving group 225.0.71.1 on interface iptv
Aug 29 18:44:23 Pamaro-FW udm-iptv[4064]: Removing MFC: 213.75.167.58 -> 224.3.2.6, InpVIf: 1
Aug 29 18:44:23 Pamaro-FW udm-iptv[4064]: Leaving group 224.3.2.6 on interface iptv
Aug 29 18:44:23 Pamaro-FW udm-iptv[4064]: Removing MFC: 217.166.226.134 -> 224.0.252.134, InpVIf: 1
Aug 29 18:44:23 Pamaro-FW udm-iptv[4064]: Leaving group 224.0.252.134 on interface iptv
Aug 29 18:44:23 Pamaro-FW udm-iptv[4064]: All routes removed. Routing table is empty.
Aug 29 18:44:23 Pamaro-FW udm-iptv[4064]: Shutdown complete....
Aug 29 18:44:23 Pamaro-FW systemd[1]: udm-iptv.service: Succeeded.
Aug 29 18:44:23 Pamaro-FW systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Aug 29 18:44:23 Pamaro-FW systemd[1]: udm-iptv.service: Consumed 1min 49.788s CPU time.
Aug 29 18:44:23 Pamaro-FW systemd[1]: Started IPTV support for the UniFi Dream Machine.
Aug 29 18:44:23 Pamaro-FW udm-iptv[15441]: Device iptv already exists.. Deleting device
Aug 29 18:44:23 Pamaro-FW udm-iptv[15441]: Obtaining IP address for VLAN interface
Aug 29 18:44:23 Pamaro-FW udm-iptv[15441]: udhcpc (v1.22.1) started
Aug 29 18:44:23 Pamaro-FW udm-iptv[15441]: Sending discover...
Aug 29 18:44:26 Pamaro-FW udm-iptv[15441]: Sending discover...
Aug 29 18:44:29 Pamaro-FW udm-iptv[15441]: Sending discover...
Aug 29 18:44:32 Pamaro-FW udm-iptv[15441]: udhcpc failed to get a DHCP lease
Aug 29 18:44:32 Pamaro-FW udm-iptv[15441]: No lease, forking to background
Aug 29 18:44:32 Pamaro-FW udm-iptv[15441]: NATing IPTV network ranges (if necessary)
Aug 29 18:44:32 Pamaro-FW udm-iptv[15441]: Setting up igmpproxy
Aug 29 18:44:32 Pamaro-FW udm-iptv[15441]: Starting igmpproxy
Aug 29 18:44:32 Pamaro-FW udm-iptv[15441]: adding VIF, Ix 0 Fl 0x0 IP 0x0103a8c0 br6, Threshold: 1, Ratelimit: 0
Aug 29 18:44:32 Pamaro-FW udm-iptv[15441]: There must be at least 1 Vif as upstream.
Aug 29 18:44:32 Pamaro-FW systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Aug 29 18:44:32 Pamaro-FW systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Aug 29 18:44:32 Pamaro-FW systemd[1]: udm-iptv.service: Consumed 114ms CPU time.
Aug 29 18:44:37 Pamaro-FW systemd[1]: udm-iptv.service: Service RestartSec=5s expired, scheduling restart.
Aug 29 18:44:37 Pamaro-FW systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 1.
Aug 29 18:44:37 Pamaro-FW systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Aug 29 18:44:37 Pamaro-FW systemd[1]: udm-iptv.service: Consumed 114ms CPU time.
Aug 29 18:44:37 Pamaro-FW systemd[1]: Started IPTV support for the UniFi Dream Machine.
Aug 29 18:44:37 Pamaro-FW udm-iptv[16081]: Device iptv already exists.. Deleting device
Aug 29 18:44:37 Pamaro-FW udm-iptv[16081]: Obtaining IP address for VLAN interface
Aug 29 18:44:37 Pamaro-FW udm-iptv[16081]: udhcpc (v1.22.1) started
Aug 29 18:44:37 Pamaro-FW udm-iptv[16081]: Sending discover...
Aug 29 18:44:40 Pamaro-FW udm-iptv[16081]: Sending discover...
Aug 29 18:44:44 Pamaro-FW udm-iptv[16081]: Sending discover...
